src/CVRF/cvrf_priv.c

```diff
@@ -0,0 +1,2292 @@
+			struct cvrf_product_status *stat = cvrf_product_status_parse(reader);
+			if (stat != NULL)
+				cvrf_vulnerability_add_cvrf_product_status(vuln, stat);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_THREATS)) {
+			cvrf_parse_container(reader, vuln->threats, CVRF_THREAT);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_CVSS_SCORE_SETS)) {
+			cvrf_parse_container(reader, vuln->score_sets, CVRF_SCORE_SET);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_REMEDIATIONS)) {
+			cvrf_parse_container(reader, vuln->remediations, CVRF_REMEDIATION);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_REFERENCES)) {
+			cvrf_parse_container(reader, vuln->references, CVRF_REFERENCE);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_ACKNOWLEDGMENTS)) {
+			cvrf_parse_container(reader, vuln->acknowledgments, CVRF_ACKNOWLEDGMENT);
+		}
+		xmlTextReaderNextNode(reader);
+	}
+	xmlTextReaderNextNode(reader);
+	return vuln;
+}
+
+struct cvrf_score_set *cvrf_score_set_parse(xmlTextReaderPtr reader) {
+	__attribute__nonnull__(reader);
+
+	struct cvrf_score_set *score_set = cvrf_score_set_new();
+	xmlTextReaderNextElement(reader);
+	while (xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_SCORE_SET) != 0) {
+		if (xmlTextReaderNodeType(reader) != XML_READER_TYPE_ELEMENT) {
+			xmlTextReaderNextNode(reader);
+			continue;
+		}
+
+		if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_VECTOR)) {
+			score_set->vector = oscap_element_string_copy(reader);
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_PRODUCT_ID)) {
+			oscap_stringlist_add_string(score_set->product_ids, oscap_element_string_copy(reader));
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_BASE_SCORE)) {
+			cvrf_score_set_add_metric(score_set, CVSS_BASE, oscap_element_string_copy(reader));
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_ENVIRONMENTAL_SCORE)) {
+			cvrf_score_set_add_metric(score_set, CVSS_ENVIRONMENTAL, oscap_element_string_copy(reader));
+		} else if (!xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_TEMPORAL_SCORE)) {
+			cvrf_score_set_add_metric(score_set, CVSS_TEMPORAL, oscap_element_string_copy(reader));
+		}
+		xmlTextReaderNextNode(reader);
+	}
+	xmlTextReaderNextNode(reader);
+	return score_set;
+}
+
+struct cvrf_involvement *cvrf_involvement_parse(xmlTextReaderPtr reader) {
+	__attribute__nonnull__(reader);
+
+	struct cvrf_involvement *involve = cvrf_involvement_new();
+	involve->status = cvrf_item_parse_type_attribute(reader, CVRF_INVOLVEMENT);
+	involve->party = cvrf_doc_publisher_type_parse(reader, "Party");
+	xmlTextReaderNextNode(reader);
+
+	while (xmlStrcmp(xmlTextReaderConstLocalName(reader), TAG_INVOLVEMENT) != 0) {
+		if (xmlTextReaderNodeType(reader) != XML_READER_TYPE_ELEMENT) {
+			xmlTextReaderNextNode(reader);
+			continue;
+		}
+		if (!xmlStrcmp(xmlTex
```

Comment:
place parsing and serializing functions of the same struct close to each other, it will make it easier to verify the whole serialization roundtrip works.